backend/src/routes/userRoute.ts

```diff
@@ -0,0 +1,66 @@
+import express, { Request, Response } from "express";
+import admin, { ServiceAccount } from "firebase-admin";
+
+import * as serviceAccount from "../firebase/ServiceAccountKey.json";
+import User, { UserDocument } from "../models/User";
+
+// Define the type for req.body
+type CreateUserRequestBody = {
+  name: string;
+  gender: string;
+  accountType: "admin" | "team";
+  approvalStatus: string; // NOTE Should this be restricted to certain values?
+  email: string;
+  password: string;
+};
+
+const router = express.Router();
+
+// Initialize Firebase Admin SDK
+admin.initializeApp({
+  credential: admin.credential.cert(serviceAccount as ServiceAccount), // Type assertion
+  databaseURL: process.env.FIREBASE_DATABASE_URL,
+});
+
+router.use(express.json());
+
+router.post(
+  "/createUser",
+  async (
+    req: Request<Record<string, never>, Record<string, never>, CreateUserRequestBody>,
+    res: Response,
+  ) => {
+    try {
+      const { name, gender, accountType, approvalStatus, email, password } = req.body;
+
+      // Create user in Firebase
+      const userRecord = await admin.auth().createUser({
+        email,
+        password,
+      } as admin.auth.CreateRequest); // Type assertion
+
+      // Set custom claim for accountType (“admin” | “team”)
+      await admin.auth().setCustomUserClaims(userRecord.uid, { accountType });
+
+      // Create user in MongoDB
+      const newUser = new User({
+        name,
+        gender,
+        accountType,
+        approvalStatus,
+        firebaseUid: userRecord.uid,
+      } as UserDocument); // Type assertion
+
+      await newUser.save();
+
+      res.status(201).json({ message: "User created successfully" });
+    } catch (error) {
+      console.error(error);
```

Comment:
We have a custom error handler [here](https://github.com/TritonSE/PIA-Program-Manager/blob/main/backend/src/errors/handler.ts). Lets use that instead of handling errors here. We can configure the error handler in app.ts like [this](https://github.com/TritonSE/ALUM-Mobile-Application/blob/main/backend/src/app.ts#L39)